mmdet/structures/bbox/bbox_overlaps.py

```diff
@@ -148,9 +148,9 @@ def bbox_overlaps(bboxes1, bboxes2, mode='iou', is_aligned=False, eps=1e-6):
         else:
             return bboxes1.new(batch_shape + (rows, cols))
 
-    area1 = (bboxes1[..., 2] - bboxes1[..., 0]) * (
+    area1 = (bboxes1[..., 2] - bboxes1[..., 0]).float() * (
```

Comment:
Is it appropriate to modify this place directly? My suggestion is to handle it in the detr head

Reply:
https://github.com/open-mmlab/mmdetection/blob/ecac3a77becc63f23d9f6980b2a36f86acd00a8a/mmdet/models/dense_heads/detr_head.py#L405-L409

I agree with you. How about casting predictions to fp32 in these lines?

Reply:
When I cast pred_instances.bbox_pred to FP32, I found the NaN Loss.
I'm still trying to figure this out.

Reply:
Hi, @hhaAndroid . I think the best way to prevent overflow is to computer `bbox_overlaps` with normalized coordinates.
How about adding options for normalization in 'GIoULoss' and 'IoUCost'?

Reply:
In this case, the changes are relatively large and may not be desirable

Reply:
But, normalizing bboxes does not affect the iou scores.
I think it can be used as a user option to maintain current logic.
Am I missing something? 

```python
import torch
from mmdet.structures.bbox import bbox_overlaps

a = torch.tensor([[[100., 200., 140., 500.]]])
b = torch.tensor([[[93.2234, 215.32, 139.99, 500.1111]]])
for mode in ['iou', 'iof', 'giou']:
    print(bbox_overlaps(a,b, mode=mode), bbox_overlaps(a/100.,b/100., mode=mode))

>>> tensor([[[0.8170]]]) tensor([[[0.8170]]])
>>> tensor([[[0.9487]]]) tensor([[[0.9487]]])
>>> tensor([[[0.8096]]]) tensor([[[0.8096]]])
```

Reply:
Yes, no effect for now.

Reply:
We should modify https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/models/task_modules/assigners/match_cost.py#L167